src/main/java/cceclipseplugin/core/PluginManager.java

```diff
@@ -226,10 +229,12 @@ private void registerNotificationHooks() {
 			long resId = notification.getResourceID();
 			ProjectRevokePermissionsNotification n = ((ProjectRevokePermissionsNotification) notification.getData());
 			Project project = storage.getProjectById(resId);
-			if (project.getPermissions() == null) {
-				project.setPermissions(new HashMap<>());
+			if (project != null) {
+				if (project.getPermissions() == null) {
+					project.setPermissions(new HashMap<>());
+				}
+				project.getPermissions().remove(n.revokeUsername);
 			}
```

Comment:
Shouldn't most of this stuff be moved to the else statement of `if (storage.getUsername().equals(n.revokeUsername))`? The `project` object should really only be modified if you're not the person being removed.